File: src/main/java/pt/joaomneto/titancompanion/adventurecreation/impl/fragments/awf/AWFAdventureCreationSuperpowerFragment.java
package pt.joaomneto.titancompanion.adventurecreation.impl.fragments.awf;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import pt.joaomneto.titancompanion.R;
import android.support.v4.app.Fragment;
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner;
import pt.joaomneto.titancompanion.adventurecreation.AdventureCreation;
import pt.joaomneto.titancompanion.adventurecreation.impl.AWFAdventureCreation;

public class AWFAdventureCreationSuperpowerFragment extends Fragment {

    public AWFAdventureCreationSuperpowerFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        View rootView = inflater.inflate(R.layout.fragment_17awf_adventure_creation_superpowers, container, false);

        List<Map<String, String>> skillList = new ArrayList<Map<String, String>>();


        final String[] stringArray = getResources().getStringArray(R.array.superpower_list);

        for (String string : stringArray) {
            Map<String, String> skill = new HashMap<String, String>();
            skill.put("skill", string);
            skillList.add(skill);
        }

        SimpleAdapter mAdapter = new SimpleAdapter(getActivity(), skillList, R.layout.potions_item, new String[]{"skill"}, new int[]{R.id.potion_name});

        ListView lView = rootView.findViewById(R.id.superpowerList);
        lView.setAdapter(mAdapter);
        lView.setSelector(R.drawable.row_selector);

        lView.setOnItemClickListener(new AdapterView.OnItemClickListener() {

            @Override
            public void onItemClick(AdapterView<?> parent, final View view, int position, long id) {
                ((AWFAdventureCreation) getActivity()).setSuperPower(stringArray[position]);
            }

        });

        return rootView;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        view.findViewById(R.id.buttonSaveAdventure).setOnClickListener((View v) -> {
            ((AdventureCreation) this.getActivity()).saveAdventure();
        });
    }


}

File: src/main/java/pt/joaomneto/titancompanion/adventurecreation/impl/fragments/awf/AWFAdventureCreationSuperpowerFragment.kt
package pt.joaomneto.titancompanion.adventurecreation.impl.fragments.awf

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ListView
import android.widget.SimpleAdapter
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventurecreation.AdventureCreation
import pt.joaomneto.titancompanion.adventurecreation.impl.AWFAdventureCreation
import java.util.ArrayList
import java.util.HashMap

class AWFAdventureCreationSuperpowerFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreate(savedInstanceState)
        val rootView = inflater.inflate(R.layout.fragment_17awf_adventure_creation_superpowers, container, false)

        val skillList = ArrayList<Map<String, String>>()

        val stringArray = resources.getStringArray(R.array.superpower_list)

        for (string in stringArray) {
            val skill = HashMap<String, String>()
            skill["skill"] = string
            skillList.add(skill)
        }

        val mAdapter = SimpleAdapter(
            activity,
            skillList,
            R.layout.potions_item,
            arrayOf("skill"),
            intArrayOf(R.id.potion_name)
        )

        val lView = rootView.findViewById<ListView>(R.id.superpowerList)
        lView.adapter = mAdapter
        lView.setSelector(R.drawable.row_selector)

        lView.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id -> (activity as AWFAdventureCreation).superPower = stringArray[position] }

        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        view.findViewById<View>(R.id.buttonSaveAdventure).setOnClickListener { v: View -> (this.activity as AdventureCreation).saveAdventure() }
    }
}
